src/cpp/core/http/ProxyUtils.cpp

```diff
@@ -100,7 +100,14 @@ bool ProxyUtils::shouldProxy(const std::string& address,
    return true;
 }
 
-const ProxyUtils& proxyUtils()
+void ProxyUtils::addNoProxyRule(const std::string& address,
```

Comment:
I would prefer we change this function to take single `rule` parameter since `createNoProxyRule` can create other types of NoProxyRules beyond address:port, _or_ change the function name to something like `addNoProxyRuleAddress` and have it create a `NoProxyRuleAddress` object directly since we don't need to parse the rule string.

Could you also add a few unit tests for this function to `ProxyUtilsTests.cpp`?

Reply:
Done; I took the first approach but also added an overload to create proxy rules from an address + port pair (so users don't have to "manually" add the colon separator when calling into the API)